lib/client.js

```diff
@@ -1942,6 +1942,16 @@ function doInitialSync(client, historyLen, includeArchived) {
                     room.addReceipt(receipts[j]);
                 }
 
+                var privateUserData = data.rooms[i].private_user_data || [];
+                for (j = 0; j < privateUserData.length; j++) {
+                    var event = _PojoToMatrixEventMapper(client)(privateUserData[j]);
```

Comment:
It took me longer than I'd like to grok this. We generally tend to use the idiom:
```javascript
utils.map(eventsWhichAreObjects, _PojoToMatrixEventMapper(client));
```

This does incur an extra loop to then do stuff with the events, but profiling hasn't shown this to be the cause of any slowness in the JS SDK.